File: src/main/java/com/papilertus/gui/modal/ModalRegistry.java
package com.papilertus.gui.modal;

import java.util.HashMap;

public class ModalRegistry {
    private static final HashMap<String, DiscordModal> modals = new HashMap<>();

    public static void registerModal(String id, DiscordModal button) {
        modals.put(id, button);
    }

    public static void unregisterModal(String id) {
        modals.remove(id);
    }

    public static DiscordModal getModalById(String id) {
        return modals.get(id);
    }
}

File: src/main/kotlin/com/papilertus/gui/modal/ModalRegistry.kt
package com.papilertus.gui.modal

object ModalRegistry {
    private val modals = HashMap<String, DiscordModal>()
    @JvmStatic
    fun registerModal(id: String, button: DiscordModal) {
        modals[id] = button
    }

    @JvmStatic
    fun unregisterModal(id: String) {
        modals.remove(id)
    }

    @JvmStatic
    fun getModalById(id: String): DiscordModal? {
        return modals[id]
    }
}